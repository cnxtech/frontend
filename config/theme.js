import theme from '@emcasa/ui'
import {fas} from '@fortawesome/pro-solid-svg-icons'
import {fal} from '@fortawesome/pro-light-svg-icons'
import icons from './icons'

export default {
  ...theme,
  icons: {emcasa: icons, default: fas, light: fal}
}
