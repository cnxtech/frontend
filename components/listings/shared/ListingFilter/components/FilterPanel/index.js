import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import theme from '@emcasa/ui'
import Row from '@emcasa/ui-dom/components/Row'
import Button from '@emcasa/ui-dom/components/Button'
import { isMobile } from 'lib/mobile'
import {
  Container
} from './styles'

class FilterPanel extends PureComponent {
  render() {
    let left = 0, top = 0
    const { title, show, panelPosition } = this.props
    if (panelPosition) {
      left = isMobile() ? theme.space[4] : panelPosition.left
      top = panelPosition.top
    }
    return (
      <Container elevation={4} p={2} show={show} left={left} top={top}>
        {isMobile() && <Row>{title}</Row>}
        <Row>
          {this.props.children}
        </Row>
        <Row justifyContent="space-between" mt={2}>
          <Button p={0} link style={{color: theme.colors.dark, height: 32}} onClick={this.props.clear}>Limpar</Button>
          <Button p={0} link style={{height: 32}} onClick={this.props.apply}>Aplicar</Button>
        </Row>
      </Container>
    )
  }
}

FilterPanel.propTypes = {
  show: PropTypes.bool.isRequired,
  apply: PropTypes.func.isRequired,
  clear: PropTypes.func.isRequired,
  panelPosition: PropTypes.object.isRequired
}

export default FilterPanel
