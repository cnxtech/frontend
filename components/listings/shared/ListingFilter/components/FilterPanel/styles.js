import styled from 'styled-components'
import theme from '@emcasa/ui'
import View from '@emcasa/ui-dom/components/View'
import Row from '@emcasa/ui-dom/components/Row'
import Button from '@emcasa/ui-dom/components/Button'

const OFFSET_TOP = 26
const WRAPPER_PADDING = theme.space[4] * 2
const MAX_FILTER_PANEL_DESKTOP_WIDTH = 300

const Container = styled(View)`
  position: absolute;
  display: ${({show}) => show ? 'block' : 'none'};
  top: ${({top}) => top ? `calc(${top}px - ${OFFSET_TOP}px)` : '0'};
  left: ${({left}) => left ? `${left}px` : '0'};
  border-radius: 4px;
  box-shadow: 0px 0px 8px -1px rgba(0,0,0,0.2);

  @media screen and (max-width: ${theme.breakpoints[0]}) {
    position: fixed;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    margin: 0;
    padding: 0;
  }
`

const Wrapper = styled(View)`
  position: relative;
  z-index: 2;
  padding: ${theme.space[2]}px;
  background-color: ${theme.colors.white};
  border-radius: 4px;

  @media screen and (max-width: ${theme.breakpoints[0]}) {
    position: relative;
    height: 100%;
    padding: ${theme.space[4]}px;
  }
`

const FilterOptions = styled(Row)`
  justify-content: flex-start;
`

const ActionsWrapper = styled(Row)`
  justify-content: space-between;
  margin-top: ${theme.space[2]}px;
  width: 100%;
  @media screen and (max-width: ${theme.breakpoints[0]}) {
    position: absolute;
    margin: 0;
    bottom: calc(20px + ${WRAPPER_PADDING}px);
    width: calc(100% - ${WRAPPER_PADDING}px);
  }

  Button {
    font-size: ${theme.fontSizes[1]}px;
  }
`

const MobileContent = styled(Row)`
  display: none;
  padding: 0;
  margin: 0;
  @media screen and (max-width: ${theme.breakpoints[0]}) {
    display: block;
  }
`

const DesktopContent = styled(Row)`
  @media screen and (max-width: ${theme.breakpoints[0]}) {
    display: none;
    padding: 0;
    margin: 0;
  }
`

export {
  Container,
  Wrapper,
  ActionsWrapper,
  FilterOptions,
  MobileContent,
  DesktopContent,
  MAX_FILTER_PANEL_DESKTOP_WIDTH
}
