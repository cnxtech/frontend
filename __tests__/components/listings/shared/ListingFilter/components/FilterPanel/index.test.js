import React from 'react'
import FilterPanel from 'components/listings/shared/ListingFilter/components/FilterPanel'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import theme from '@emcasa/ui'
import { ThemeProvider } from 'styled-components'

describe('<FilterPanel/>', () => {
  it('should render component', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <FilterPanel
          show={true}
          close={() => {}}
          apply={() => {}}
          clear={() => {}}
          panelPosition={{left: 0, top: 0}}
        />
      </ThemeProvider>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
