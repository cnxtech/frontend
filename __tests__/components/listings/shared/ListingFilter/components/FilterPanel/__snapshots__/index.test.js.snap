// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterPanel/> should render component 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c9 {
  width: 33.33333333333333%;
}

.c10 {
  outline: none;
  display: block;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #e0e6ed;
  background-color: #ffffff;
  border-radius: 4px;
  border-width: 1px;
  width: 100%;
  height: 60px;
  padding: 0px;
  color: #3c4858;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  color: #3c4858;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
}

.c10 :hover {
  background-color: #ffffff;
}

.c11 {
  outline: none;
  display: block;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #f50057;
  background-color: #f50057;
  border-radius: 4px;
  border-width: 1px;
  width: 100%;
  height: 60px;
  padding: 0px;
  color: #ffffff;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
}

.c11 :hover {
  background-color: #f50057;
}

.c13 {
  outline: none;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #e0e6ed;
  background-color: #ffffff;
  border-radius: 0;
  border-width: 0;
  width: auto;
  height: 32px;
  padding: 0px;
  color: #f50057;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  color: #3c4858;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
}

.c13 :hover {
  background-color: #ffffff;
}

.c14 {
  outline: none;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #e0e6ed;
  background-color: #ffffff;
  border-radius: 0;
  border-width: 0;
  width: auto;
  height: 32px;
  padding: 0px;
  color: #f50057;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
}

.c14 :hover {
  background-color: #ffffff;
}

.c6 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #3c4858;
  font-size: 18px;
}

.c1 {
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

.c0 {
  position: absolute;
  display: block;
  top: 0;
  left: 0;
  padding: 10px;
  background-color: #ffffff;
  width: 300px;
}

.c7 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c8 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 10px;
  width: 100%;
}

.c3 {
  display: none;
  padding: 0;
  margin: 0;
}

@media screen and (max-width:780px) {
  .c0 {
    position: fixed;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    margin: 0;
    padding: 0;
  }
}

@media screen and (max-width:780px) {
  .c2 {
    position: relative;
    height: 100%;
    padding: 20px;
  }
}

@media screen and (max-width:780px) {
  .c8 {
    position: absolute;
    margin: 0;
    bottom: calc(20px + 40px);
    width: calc(100% - 40px);
  }
}

@media screen and (max-width:780px) {
  .c3 {
    display: block;
  }
}

@media screen and (max-width:780px) {
  .c12 {
    display: none;
    padding: 0;
    margin: 0;
  }
}

<div
  className="c0 c1"
  elevation={4}
>
  <div
    className="c2 "
  >
    <div
      className="c3 c4"
    >
      <div
        className="c5"
        onClick={[Function]}
        width={1}
      >
        <div
          className=""
        >
          <svg
            height={20}
            version="1.1"
            viewBox="0 0 352 512"
            x={0}
            y={0}
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              name="times"
              type="default"
            />
          </svg>
        </div>
      </div>
      <div
        className="c4"
      >
        <p
          className="c6"
          color="dark"
          fontSize="default"
        />
      </div>
    </div>
    <div
      className="c7 c4"
    />
    <div
      className="c3 c4"
    >
      <div
        className="c8 c4"
      >
        <div
          className="c9"
          width={0.3333333333333333}
        >
          <button
            className="c10"
            color="dark"
            fontSize="default"
            height="tall"
            onClick={[Function]}
          >
            Limpar
          </button>
        </div>
        <div
          className="c9"
          style={
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
            }
          }
          width={0.3333333333333333}
        >
          <button
            className="c11"
            fontSize="default"
            height="tall"
            onClick={[Function]}
          >
            Aplicar
          </button>
        </div>
      </div>
    </div>
    <div
      className="c12 c4"
    >
      <div
        className="c8 c4"
      >
        <div
          className="c9"
          width={0.3333333333333333}
        >
          <button
            className="c13"
            color="dark"
            fontSize="default"
            height="short"
            onClick={[Function]}
          >
            Limpar
          </button>
        </div>
        <div
          className="c9"
          style={
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
            }
          }
          width={0.3333333333333333}
        >
          <button
            className="c14"
            fontSize="default"
            height="short"
            onClick={[Function]}
          >
            Aplicar
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
